### ISSUE TITLE
jax2tf shape polymorphism does not work with BatchNorm

### ISSUE THREAD

**Issue by gnecula** [google/jax#6973]
The root cause is a code snippet of the following form:

```
# x: f32[b, 8]
def f_jax(x): 
   return jnp.mean(x)

jax2tf.convert(f_jax, polymorphic_shapes=["b, ..."])(np.ones((3, 4)))
```

The issue is that `jnp.mean` tries to compute and divide by the size of `x`, which is not a constant.
